core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -74,8 +74,9 @@ class StorageStatusListener(conf: SparkConf) extends SparkListener {
     synchronized {
       val blockManagerId = blockManagerAdded.blockManagerId
       val executorId = blockManagerId.executorId
-      val maxMem = blockManagerAdded.maxMem
-      val storageStatus = new StorageStatus(blockManagerId, maxMem)
+      val maxOnHeapMem = blockManagerAdded.maxOnHeapMem.getOrElse(blockManagerAdded.maxMem)
+      val maxOffHeapMem = blockManagerAdded.maxOffHeapMem.getOrElse(0L)
```

Comment:
the only time these fields will be `None` is when reading from old event logs, right?  If so, its worth adding that in as a comment above this.